app/controllers/application_controller.rb

```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+private
+
+    def check_token
+      hashtext = JSON.parse(params.first[0])
+      user = User.find_by(google_credentials: hashtext['google_credentials'])
+      unless user.chrome_token == hashtext['chrome_token']
+        render status: 401
```

Comment:
Exactly, if you render or return something in that method the action is not
executed. And this method only runs before the actions that are specified
in each controller.

Jorge Rubio C.
On Mar 11, 2015 2:55 PM, "Luke Thomas" <notifications@github.com> wrote:

> In app/controllers/application_controller.rb
> <https://github.com/echoThat/echo_that_api/pull/13#discussion_r26243651>:
>
> > @@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
> >    # Prevent CSRF attacks by raising an exception.
> >    # For APIs, you may want to use :null_session instead.
> >    protect_from_forgery with: :exception
> > +
> > +private
> > +
> > +    def check_token
> > +      hashtext = JSON.parse(params.first[0])
> > +      user = User.find_by(google_credentials: hashtext['google_credentials'])
> > +      unless user.chrome_token == hashtext['chrome_token']
> > +        render status: 401
>
> Does "render status" act as a break or return value? In other words, does
> it interrupt the EchosController#create cycle?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/echoThat/echo_that_api/pull/13/files#r26243651>.
>